cloud-platform/add-secrets-to-deployment.md

```diff
@@ -0,0 +1,232 @@
+---
+category: cloud-platform
+expires: 2018-01-31
+---
+# Adding a secret to an application
+
+## Overview
+
+The aim of this guide is to walkthrough the process of adding a secret (in this example for aws access-key credentials) to a previously deployed application in the Cloud Platform.
+
+## Prerequisites
+
+This guide assumes the following:
+
+* You have previously set up an env. See [Creating a Cloud Platform Environment](/cloud-platform/env-create)
+* You have previously deployed your application. See [Deploying an application to the Cloud-Platform](/cloud-platform/app-deploy)
+* Check your deployment is running. See [Interacting with the application](/cloud-platform/app-deploy/#interacting-with-the-application)
+
+## Configuring secrets
+
+The following is an example of encoding (configuring) aws access-key credentials in your deployment.
+See [kuberenetes using secrets as environment variables](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables)
+
+for detailed information regarding providing base64 values in secret objects to Kuberenetes pods)
+
+Create your AWS Credentials access key (making a note of the aws_access_key_id and aws_secret_access_key)
+
+### base64-encode your secret as follows:
+
+In this example  aws_access_key_id is 'AKIAFTKSAW15HJLOGD'. Issue the following command to base64-encode:
+
+```
+echo -n 'AKIAFTKSAW15HJLOGD' | base64 -b0
+```
+
+This will return the encoded id 'QUtJQUZUS1NBVzE1SEpMT0dE'
+
+In this example the is aws_secret_access_key 'g8hjpmhvgfhk4547gfdshhjj'. Issue the following command to base64-encode:
+
+```
+echo -n 'QUtJQUZUS1NBVzE1SEpMT0dE' | base64 -b0
+```
+
+This will return the encoded secret 'UVV0SlFVWlVTMU5CVnpFMVNFcE1UMGRF' 
+
+## Creating the secret
+
+There are 2 ways of creating the secret:
+
+1. create a generic secret - for Concourse's git-crypt key
```

Comment:
Users aren't going to be using Concourse for now, so I don't think it's worth mentioning it at the moment.